examples/image_captioning/train.py

```diff
@@ -0,0 +1,141 @@
+import argparse
+
+import chainer
+from chainer.datasets import TransformDataset
+from chainer import iterators
+from chainer import optimizers
+from chainer import training
+from chainer.training import extensions
+
+import datasets
+from model import ImageCaptionModel
+
+
+class TestModeEvaluator(extensions.Evaluator):
+
+    """Evaluates the model on the validation dataset."""
+
+    def evaluate(self):
+        with chainer.using_config('train', False), \
+                chainer.no_backprop_mode():
+            ret = super(TestModeEvaluator, self).evaluate()
+        return ret
```

Comment:
This `TestModeEvaluator` may not required, in the view of 
`chainer.using_config('train', False)` and `chainer.no_backprop_mode()`.
These are performed even in the common `extensions.Evaluator`.
https://github.com/chainer/chainer/blob/v3.2.0/chainer/training/extensions/evaluator.py#L136-L137
https://github.com/chainer/chainer/blob/v3.2.0/chainer/training/extensions/evaluator.py#L182

Or, you have other idea about this class?

Reply:
Thank you for pointing that out! Removed the example-defined class!